src/View/ImageFusion/ImageFusionWindow.py

```diff
@@ -350,105 +347,173 @@ def search_progress(self, progress_update):
+        is_expanded = False \
+            if (item.isExpanded() is True and
+                item.checkState(0) == Qt.CheckState.Unchecked) else True
+        self.display_a_tree_branch(item, is_expanded)
+
         selected_patient = item
-        # If the item is not top-level, bubble up to see which top-level
-        # item this item belongs to
+        # If the item is not top-level, bubble up to see which top-level item
+        # this item belongs to
         if self.open_patient_window_patients_tree.invisibleRootItem(). \
                 indexOfChild(item) == -1:
             while self.open_patient_window_patients_tree.invisibleRootItem(). \
                     indexOfChild(selected_patient) == -1:
                 selected_patient = selected_patient.parent()
 
         # Uncheck previous patient if a different patient is selected
-        if item.checkState(0) == Qt.CheckState.Checked and \
-                self.last_patient != selected_patient:
+        if item.checkState(0) == Qt.CheckState.Checked and self.last_patient \
+                != selected_patient:
             if self.last_patient is not None:
-                self.last_patient.setCheckState(0, Qt.CheckState.Unchecked)
-                self.last_patient.setSelected(False)
+                last_patient_checked_items = self.get_checked_nodes(
+                    self.last_patient)
+                for checked_item in last_patient_checked_items:
+                    checked_item.setCheckState(0, Qt.Unchecked)
             self.last_patient = selected_patient
 
-        # Get the types of all selected leaves
-        selected_series_types = set()
-        selected_studies_names = set()
-        for checked_item in self.get_checked_leaves():
-            selected_studies_names.add(checked_item.parent().text(0))
-            series_type = checked_item.dicom_object.get_series_type()
-            if type(series_type) == str:
-                selected_series_types.add(series_type)
-            else:
-                selected_series_types.update(series_type)
-
-        # Check if multiple studies are selected
-        if len(selected_studies_names) == 1 & \
-                len(list({'CT', 'MR', 'PT'} & selected_series_types)) != 0:
-            self.open_patient_window_confirm_button.setDisabled(False)
-        elif len(selected_studies_names) > 1:
-            header = "Only one study can be opened."
-            self.open_patient_window_confirm_button.setDisabled(True)
-
-        # Check the existence of IMAGE, RTSTRUCT and RTDOSE files
-        if len(list({'CT', 'MR', 'PT'} & selected_series_types)) == 0:
-            header = "Cannot proceed without an image file."
-            self.open_patient_window_confirm_button.setDisabled(True)
+        # Check selected items and display warning messages
+        self.check_selected_items(selected_patient)
+
+    def display_a_tree_branch(self, node, is_expanded):
+        """ Expand/Close a tree branch """
```

Comment:
Docstring is incomplete, parameter definitions are needed.

Reply:
Could Team 23 advise what is 'is_expanded' meant to be?

Reply:
I can't say for sure but it looks like this variable is used to set whether a node on the tree (when selecting a patient/image set for image fusion) is expanded on not.

Reply:
I didn't write this code. I can't comment - I also don't want to assume. Can team 23 please fix their docstring as well as their own function within the next PR, thanks.

Reply:
@peteqian i think this is from last year's team... Regardless, it is a boolean parameter to mark whether a node 's information is expanded. If you can update the docstring here for us would be great. Thank you.

Reply:
This isn't last year's code. This code was developed by your team in this [PR](https://github.com/didymo/OnkoDICOM/pull/168/files). I will make the fix regarding my file, however, upon investigating this flag and how it sets - shorthand notation makes it difficult to read if the flag is constantly set to False or True due to the PEP-8 Compliance.

`
is_expanded = False \
            if (item.isExpanded() is True and
                item.checkState(0) == Qt.CheckState.Unchecked) else True
        self.display_a_tree_branch(item, is_expanded)
`

Also, a suggestion might be to is to remove the "is True" section of the if condition, as this makes the condition require an additional useless parameter for evaluating Truths.

Reply:
Yep, thanks for pointing that out, and sorry I thought it was last year's code 😂 we will fix the condition in our next PR so that it's more readable.

Reply:
@peteqian that being said, the code in here is copied from PR #168. It's a bit concerning since it shows that from now on, if we make changes to Open Patient Window or how the directory tree is structured, we need to do the same thing for the one In Image Fusion as well. It makes it more difficult to maintain the code. I'm not saying that we need to fix the repetition now, but our team will document it for next year's aims planning so that they can work on integrating these two windows together.

Reply:
But does thou approve such a request?

Reply:
@peteqian i'm still testing the program for functionalities, so if it all works well i'll approve this PR for sure

Reply:
> @peteqian that being said, the code in here is copied from PR #168, is that correct? It's a bit concerning since it shows that from now on, if we make changes to Open Patient Window or how the directory tree is structured, we need to do the same thing for the one In Image Fusion as well. It makes it more difficult to maintain the code. I'm not saying that we need to fix the repetition now given the time constraint, but our team will document it for next year's aims planning so that they can work on integrating these two windows together.

I do agree with this. However, previously when I was trying to inherit this class - I was running into issues where the subclass (UIImageFusionWindow) was not instantiating the superclass (UIOpenPatientWindow) properly - this was most likely due to no __init__() present in the UIOpenPatientWindow class (although just speculating at this point). If you were to make the class inheritable - then refactoring would be very easy.